numpy/_core/src/multiarray/methods.c

```diff
@@ -892,24 +892,24 @@ static PyObject *
 array_wraparray(PyArrayObject *self, PyObject *args)
 {
     PyArrayObject *arr;
-    PyObject *obj;
+    PyObject *context_ignored = NULL;
+    /*
+     * return_scalar should always be passed, if it is not default to how
+     * this function behaved in older NumPy versions.
+     */
+    int return_scalar = 0;
 
-    if (PyTuple_Size(args) < 1) {
-        PyErr_SetString(PyExc_TypeError,
-                        "only accepts 1 argument");
+    if (!PyArg_ParseTuple(args, "O!|OO&:__array_wrap__",
+                &PyArray_Type, &arr, &context_ignored,
+                &PyArray_OptionalBoolConverter, &return_scalar)) {
```

Comment:
I opted to just allow passing `None` for `return_scalar`.  I don't think anyone ever should, but it feels like it doesn't hurt (and we should have the default for when nothing is passed).

Reply:
Makes sense.

p.s. Had to check what `O!` did, but like that it removes the need for raising our own `TypeError`.